dll/win32/msafd/misc/dllmain.c

```diff
@@ -2499,7 +2499,11 @@ WSPIoctl(IN  SOCKET Handle,
                     *((PVOID *)lpvOutBuffer) = WSPConnectEx;
                     cbRet = sizeof(PVOID);
                     Errno = NO_ERROR;
-                    Ret = NO_ERROR;
+                    /* See CORE-14966 and associated commits.
+                     * Original line below was 'Ret = NO_ERROR:'.
+                     * This caused winetest ws2_32:sock to hang.
+                     * This new Ret value allows the test to complete. */
+                    Ret = SOCKET_ERROR;
```

Comment:
Since this introduces a hack, I'd at least expect to see a DPRINT here, but I much prefer not adding hacks like this.